### ISSUE TITLE
cmsconnect: job monitoring

### ISSUE THREAD

**Issue by govoni** [cms-sw/genproductions#1606]
Dear experts,

I am running the production of a gridpack on CMSConnect for the first time ever and I am wandering the following:
  * can I run the production interactively and then disconnect from the site, after submitting the job, or is there some macumba needed to retain tokes for the submitted jobs?
  * how do I monitor the status of the jobs?
  * how do I know that the gridpack production is finished?

I would like to suggest that these pieces of information are added to the gen twiki page [here](https://twiki.cern.ch/twiki/bin/view/CMS/QuickGuideMadGraph5aMCatNLO#Gridpack_generation_on_the_grid) 

pietro

**Comment by perrozzi** [comment]
@khurtado

**Comment by khurtado** [comment]
Hi @govoni:

- Just submit the master in background without hangups with `nohup`, as indicated in the [twiki instructions for cms connect](https://twiki.cern.ch/twiki/bin/viewauth/CMS/QuickGuideMadGraph5aMCatNLO#Gridpack_generation_on_the_grid). No tokens to worry about, but you need to make sure your grid certificates are always alive (so, you need to renew your certificates via `voms-proxy-init -voms cms -valid 192:00` every e.g 5 or 6 days to make sure the grid keeps accepting your gridpack jobs)    
- When you run `nohup`, you redirect the output to some debug file. In the twiki example: `mysubmit_wplus4f_NLO.debug`. You can check that file to monitor the progress of the cmsconnect submit script, you will also have the logfile `wplus4f_NLO_codegen.log` and `wplus4f_NLO.log` for the CODEGEN step and INTEGRATE steps respectively. You cannot see grid jobs outputs in real-time, but the logfiles above allow you to see the status of the master, how many jobs are running, how many are pending, possible errors, etc.
- Basically, when you submit your gridpack, you will only have `mysubmit_wplus4f_NLO.debug`, showing you it will start submitting the `CODEGEN` step, once that is finished you will be able to see `wplus4f_NLO_codegen.log` with its status if it failed, the directory `condor_logs` will show you  the output and error logfiles. If `CODEGEN` went well, you will see in `mysubmit_wplus4f_NLO.debug` that the INTEGRATE step compilation is starting and you will also be able to see `wplus4f_NLO.log` to monitor the status of the INTEGRATE step.

**Comment by govoni** [comment]
@khurtado thanks a lot for the explanations, and for updating the twiki. 
Also, if I understand correctly the result of the gridpack creation gets saved in a folder named  [card_name], located in the folder from where I submitted the gridpack job, right?
If an existing folder with that name exists, it gets deleted, also. This is part of the game I guess, but maybe a check of existence of the folder or warning is needed: in my case, I put the Madgraph datacards in a folder named  [card_name], so they disappeared because of this.

**Comment by govoni** [comment]
@khurtado thanks a lot for the explanations and for updating the twiki!
